bittensor/_dataset/dataset_impl.py

```diff
@@ -169,46 +175,141 @@ def __del__(self):
+            return True
         
-        while i < self.max_datasets:
+        except Exception as E:
+            print(E)
+            logger.warning("Save failed:".ljust(20) + "<blue>{}</blue>".format(file_meta['Name']))
+            return False
+
+    def get_text(self, file_meta):
+        r""" Either load a file from disk or download it from IPFS
+        Args:
+            file_meta (dict of str: int):
+                Specify the details of the file in the format of {'Name': , 'Hash':}.
+
+        Return:
+            text (str):
+                The text that we get from the file (from disk or IPFS).     
+        """
+        response = self.get_ipfs_file(self.text_dir, file_meta)
+        if (response != None) and (response.status_code == 200):
+            text = response.text
+            self.IPFS_fails = 0
             
-            dataset_key = list(self.dataset_hashes.keys())[dataset_hashes_order[i]]
-            dataset_hash = self.dataset_hashes[dataset_key]
-            i += 1
-            logger.success("Loading dataset:".ljust(20) + "<blue>{}</blue>".format(dataset_key))
-            response = self.retrieve_directory(self.cat, (('arg', dataset_hash),))
+            if self.save_dataset and self.dataset_hashes[file_meta['Folder']]['Size'] < self.backup_dataset_cap_size:
+                self.save_hash( file_meta, text )
+                print('old file size', self.dataset_hashes[file_meta['Folder']])
+                self.dataset_hashes[file_meta['Folder']]['Size'] += file_meta['Size']
+                print('new file size', self.dataset_hashes[file_meta['Folder']])
             
-            if (response == None) or (response.status_code != 200):
-                logger.warning("Ignoring directory:".ljust(20) + "<blue>{}</blue>".format(dataset_key))
+        else:
+            logger.warning("Failed to get text".ljust(20) + "<blue>{}</blue>".format(file_meta['Name']))
+            self.IPFS_fails += 1
             
-            else:
-                # --- Get the directory links if there is valid response, else check on another dataset_hash 
-                directories += response.json()
-                logger.success("Loaded dataset:".ljust(20) + "<blue>{}</blue>".format(dataset_key))
-                
-        if len(directories) == 0:
-            directories = None
+        return text 
+
+    def get_dataset(self , file_meta):
+        r""" Either load a dataset, which is a list of hashes, from disk or download it from IPFS
+        Args:
+            file_meta (dict of str: int):
+                Specify the details of the dataset in the format of {'Name': , 'Hash':}.
+
+        Return:
+            hashes (list):
+                The hashes from the dataset downloaded from disk or IPFS.     
+        """
+        # --- Load text from path
+        logger.success( f"Getting dataset: {file_meta['Name']}" )
         
-        return directories
+        hashes = None
```

Comment:
Is `hashes = None` necessary since you're setting it in the next line? `self.load_hash` should come back with a None if it doesn't work anyways